### DISCUSSION TITLE
How to parallel execution of different processes

### DISCUSSION THREAD

**Discussion by DescartesM** [nextflow-io/nextflow#3813] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/6761716/228707602-68c43674-2ab1-4b5e-bb3b-6d874c077c0f.png)

The scene is described in the above figure; Each process has input and output, and the process B and process C use the same input from process A;

Question:
1. I don't know how to orchestrate my business process...
2. For the process D, what would be the order of his input parameters?

**Comment by mribeirodantas** [comment, marked as answer]
```Groovy
PROCESS_A(some, inputs, that, you want)
  | PROCESS_B & PROCESS_C
  | mix
  | PROCESS_D
```

Check the example below:

```Groovy
process PROCESS_A {
  input:
  val x

  output:
  val y

  exec:
  y = x*10
}

process PROCESS_B {
  input:
  val x

  output:
  val y

  when:
  x <= 50

  exec:
  y = x*10
}

process PROCESS_C {
  input:
  val x

  output:
  val y

  when:
  x > 50

  exec:
  y = x*10
}

process PROCESS_D {
  debug true

  input:
  val x

  output:
  stdout

  script:
  """
  echo $x
  """
}

workflow {
  Channel
    .of(1..10)
  | PROCESS_A
  | PROCESS_B & PROCESS_C
  | mix
  | PROCESS_D
}
```

If you don't want to use the `mix` channel operator, you can remove this line and change PROCESS_D to expect two inputs, like in the snippet below.

```Groovy
process PROCESS_D {
  debug true

  input:
  val x
  val y

  output:
  stdout

  script:
  """
  echo $x
  echo $y
  """
}
```
